ml-agents/mlagents/trainers/trainer_util.py

```diff
@@ -37,10 +36,10 @@ def __init__(
         self.meta_curriculum = meta_curriculum
         self.multi_gpu = multi_gpu
 
-    def generate(self, brain_parameters: BrainParameters) -> Trainer:
+    def generate(self, brain_name: str) -> Trainer:
         return initialize_trainer(
             self.trainer_config,
-            brain_parameters,
+            brain_name,
```

Comment:
Do we even need the brain_name here? Isn't it passed when the policy is created?

Reply:
Brain parameters are. The brain name there is used to grab config params etc.

Reply:
I might be crazy but I don't see `brain_name` being used in the Trainer

Reply:
The logger uses it to generate a path/label and __str__ uses it.  We can probably cut it out though.

Reply:
It gets used by the logger in Trainer as a label/path and also in the __str__ method

Reply:
Oh IC. OK makes sense